components/org.wso2.carbon.identity.oauth.dcr.endpoint/src/test/java/org/wso2/carbon/identity/oauth2/dcr/endpoint/util/DCRMUtilsTest.java

```diff
@@ -0,0 +1,107 @@
+import org.testng.Assert;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+import org.wso2.carbon.identity.oauth.dcr.exception.DCRMException;
+import org.wso2.carbon.identity.oauth2.dcr.endpoint.Exceptions.DCRMEndpointException;
+import org.wso2.carbon.identity.oauth2.dcr.endpoint.dto.RegistrationRequestDTO;
+import org.wso2.carbon.identity.oauth2.dcr.endpoint.dto.UpdateRequestDTO;
+
+import javax.ws.rs.core.Response;
+import java.util.ArrayList;
+import java.util.List;
+
+public class DCRMUtilsTest {
+    private List<String> redirectUris = new ArrayList<>();
+    private List<String> grantTypes = new ArrayList<>();
+    private final String client_name = "Application";
+
+    @BeforeMethod
+    public void setUp() throws Exception {
+        redirectUris.add("https://op.certification.openid.net:60845/authz_cb");
+        grantTypes.add("authorization_code");
+    }
+
+    @Test
+    public void testGetApplicationRegistrationRequest() throws Exception {
+        RegistrationRequestDTO  registrationRequestDTO = new RegistrationRequestDTO();
+        registrationRequestDTO.setClientName(client_name);
+        registrationRequestDTO.setRedirectUris(redirectUris);
+        registrationRequestDTO.setGrantTypes(grantTypes);
+        Assert.assertNotNull(DCRMUtils.getApplicationRegistrationRequest(registrationRequestDTO));
+    }
+
+    @Test
+    public void testGetApplicationUpdateRequest() throws Exception {
+        UpdateRequestDTO updateRequestDTO = new UpdateRequestDTO();
+        updateRequestDTO.setClientName(client_name);
+        updateRequestDTO.setRedirectUris(redirectUris);
+        updateRequestDTO.setGrantTypes(grantTypes);
+        Assert.assertNotNull(DCRMUtils.getApplicationUpdateRequest(updateRequestDTO));
+    }
+
+    @DataProvider(name = "BuildDCRMException")
+    public Object[][] buildDCRMException() {
+        DCRMException dcrmException1 = new DCRMException(null,"error code null");
+        DCRMException dcrmException2 = new DCRMException("CONFLICT_","error code start with conflict");
+        DCRMException dcrmException3 = new DCRMException("BAD_REQUEST_INVALID_REDIRECT_URI","error code for invalid redirect URI");
+        DCRMException dcrmException4 = new DCRMException("NOT_FOUND_","error code start with not found");
+        DCRMException dcrmException5 = new DCRMException("BAD_REQUEST_","error code start with bad request");
+        return new Object[][] {
+                {dcrmException1},
+                {dcrmException2},
+                {dcrmException3},
+                {dcrmException4},
+                {dcrmException5}
+        };
+    }
+
+    @Test(dataProvider = "BuildDCRMException", expectedExceptions = DCRMEndpointException.class)
+    public void testHandleErrorResponse(DCRMException dcrmException) throws Exception {
+        Log log =null;
+        DCRMUtils.handleErrorResponse(dcrmException, log);
```

Comment:
Can you explain what are we trying to assert/test in this test case?

Reply:
whether the expected exception is thrown by the method handleErrorResponse method.